Jenkinsfile

```diff
@@ -18,7 +18,50 @@ pipeline {
                 slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
             }
         }
-        stage('Parallel Build') {
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
+          when {
+            allOf {
+              expression { env.CHANGE_ID != null }
+              expression { env.CHANGE_TARGET != null }
+            }
+          }
+          // TODO DDF-2971 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
+          steps {
+            parallel(
+              linux: {
+                node('linux-large') {
+                  retry(3) {
+                    checkout scm
+                  }
+                  timeout(time: 3, unit: 'HOURS') {
```

Comment:
It would probably be a build smell if we ever started exceeding 3 hours for this kind of build :-)

Reply:
@shaundmorris we get pretty close sometimes: https://jenkins.codice.org/job/DDF-Linux-PR/1914/

Reply:
@oconnormi, @LinkMJB  do we think this timeout is appropriate or should we adjust or try to improve it on a later?

Reply:
Yeah i think we should leave it this way since we get close sometimes. Hopefully things will improve in the future